### PR TITLE
Prepare Azure.Core 1.16.0 release

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/core/Azure.Core/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.16.0-beta.1 (Unreleased)
+## 1.16.0 (2021-07-06)
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core/CHANGELOG.md:3]: Shouldn't this be 6/30?

File: sdk/core/Azure.Core/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core/CHANGELOG.md:4]: Are there any new features or fixes?

Comment (pakrym) [reply sdk/core/Azure.Core/CHANGELOG.md:4]: Yep, asked @christothes to add a changelog for the feature he added.

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**